sstables/metadata_collector.hh

```diff
@@ -136,6 +136,8 @@ private:
     int _sstable_level = 0;
     std::optional<position_in_partition> _min_clustering_pos;
     std::optional<position_in_partition> _max_clustering_pos;
+    std::optional<position_in_partition> _first_clustering_pos;
+    std::optional<position_in_partition> _last_clustering_pos;
```

Comment:
I see we already have `{min,max}_clustering_pos` above. What is difference compared to those? Might worth a comment to help any future readers wondering the same.

Reply:
min and max refer to value, whereas first and last refer to order, meaning first == first position in first partition, whereas last = last position in last partition. will add a comment.